content/document.js

```diff
@@ -206,16 +193,42 @@ function update(contentRoot, folder, rawHtml, metadata) {
   }
 }
 
-function del(contentRoot, folder) {
-  const { metadata } = read(contentRoot, folder);
+function del(folder) {
+  const { metadata } = read(folder, { metadata: true });
   fs.rmdirSync(folder, { recursive: true });
-  updateWikiHistory(path.join(contentRoot, metadata.locale), metadata.slug);
+  updateWikiHistory(path.join(CONTENT_ROOT, metadata.locale), metadata.slug);
+}
+
+const findByURL = memoizeDuringBuild((url, fields = null) => {
```

Comment:
This function has to be more robust in terms of its handling of the incoming `url`, since the incoming `url` can come from the arguments of macro calls (which are notoriously unreliable). It'll have to do two things:
- repair an unorthodox URL (like we do currently with https://github.com/mdn/yari/blob/674e331d2f79f3c340339f31cf80b387b2bf7e60/content/scripts/build.js#L244)
- replace a redirected URL with its new target URL

so basically it'll have to do what the current [`cleanUri`](https://github.com/mdn/yari/blob/674e331d2f79f3c340339f31cf80b387b2bf7e60/content/scripts/build.js#L369) does.

Reply:
Oh, I see you already have this functionality within `info.js` as `return Redirect.resolve(repairURL(uri));` within `getUriKey()`. I wonder if `repairURL(uri)` belongs better in `content/utils.js`?

Reply:
Actually, the more I think about this, we only really need to clean the URL's arriving from the content (e.g., arguments to macro calls), so I suppose we could just pre-wrap the `globals.renderFromURL` within `kumascript/src/environment.js`, so something like:
```js
globals.renderFromURL = (url) => renderFromURL(getUriKey(url));
```